src/snippets/snippets.ts

```diff
@@ -1,9 +1,23 @@
-export interface Snippet {
-    trigger: string,
-    replacement: string,
-    options: string,
-    priority?: number,
-    description?: string
+import { Options, parseOptions } from "./options";
+
+export class RawSnippet {
+    trigger: string;
+    replacement: string;
+    options: string | Options;
```

Comment:
Since `RawSnippet` is a direct representation of the snippet file, shouldn't `options` strictly be a string? (Or is the intent for users to be able to supply an `Options` object to specify options?)

Reply:
I also thought that, but TypeScript won't let me override the type in the derived `ParsedSnippet` class without errors.

Reply:
Ah I see. Since separating things into `RawSnippet`s and `ParsedSnippets` serves the idea that the representation of parsed snippets need not depend on that of raw snippets, maybe it'd be better to make `ParsedSnippet` its own independent class and not a derived one?

Reply:
Implemented, thanks!